### PR TITLE
Fix tabix histogram for CanvasFeatuers

### LINKED ISSUE #935: Automate GFF3Tabix Feature Density
Is it possible to show a feature density histogram once the feature density or zoom level exceed some threshold for the `GFF3Tabix` `storeClass`?

This is possible for BAM files and also works when importing GFF3 using `flatfile-to-json.pl` but I can't figure out if it's possible for `GFF3Tabix`.

### PR DESCRIPTION
**Pull request by nathandunn**
Fixes #935 

- [x] add volvox example with BigWig if not there
- [x] add volvox exmple without BigWig 
- [x] adjust feature density to sane default, confirm adjustment

- [x] allows current use of supplying alternate store such as BigWig
- [x] Provides automated feature density for GFF3Tabix versus CanvasFeatures
  - [x] implement getRegionFeatureDensities to get histograms
  - [x] auto calculate scoreMax, max
  - [x] implement getGlobal and getRegion?
- [x] provides reasonably similar densities to HTMLFeatures (or at least sane defaults)

### CODE REVIEW

**Comment by nathandunn** [comment]
http://gmod.org/wiki/JBrowse_Configuration_Guide#GET_.28base.29.2Fstats.2FregionFeatureDensities.2F.28refseq_name.29.3Fstart.3D123.26end.3D456.26basesPerBin.3D20000

**Comment by nathandunn** [comment]
screen shot 2018-01-24 at 4 34 45 pm

HTMLFeatures don't seem to picking up the scale properly put canvas scale works

**Comment by nathandunn** [comment]
Fixes #953

**Comment by nathandunn** [comment]
Fixes #935

**Comment by nathandunn** [comment]
@cmdcolin  / @rbuels  I think this is ready for review.

**Comment by nathandunn** [comment]
This also has a fix for 'addStores', which should have been done within the master branch, but it ended up here.   It is the correct fix, however.

File: src/JBrowse/QueryParamConfigMapper.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ define(['dojo/_base/declare', 'JBrowse/Util/dot-object'], function (declare, dot
             var storeTracks = {};
             var storeBookmarks = {};
             Object.keys(queryParams).forEach(function (queryParam) {
-                if (queryParam.indexOf('addStore\.') == 0) {
+                if (queryParam.indexOf('addStores\.') == 0) {
```
Comment (rbuels) [inline src/JBrowse/QueryParamConfigMapper.js]: Could you add a block comment to the top of this file saying what this class does?  For posterity.

File: src/JBrowse/Store/TabixIndexedFile.js [hunk 2]
```diff
@@ -40,6 +40,25 @@ return declare( null, {
...
 
+    countLines: function( ref, min, max, itemCallback, finishCallback, errorCallback ) {
+        var thisB = this;
+        var args = Array.prototype.slice.call(arguments);
+        this.indexLoaded.then(function() {
+            thisB._count.apply( thisB, args );
+        }, errorCallback);
+    },
+
+    _count: function( ref, min, max, itemCallback, finishCallback, errorCallback ) {
```
Comment (rbuels) [inline src/JBrowse/Store/TabixIndexedFile.js]: this function doesn't look right.  it is returning count synchronously, but it also calls an error callback.

and it's returning a count of the blocks in the range, not the lines.  what's up here?

File: src/JBrowse/QueryParamConfigMapper.js (hunk 1 above)
Comment (nathandunn) [reply src/JBrowse/QueryParamConfigMapper.js]: added doc at the top.  Thanks.

File: src/JBrowse/Store/TabixIndexedFile.js (hunk 2 above)
Comment (nathandunn) [reply src/JBrowse/Store/TabixIndexedFile.js]: good idea.  I removed it as they weren't actually used.

**Comment by nathandunn** [comment]
@rbuels  Ready for re-review assuming that everything passes.

**Pull request merged**